torch/_functorch/aot_autograd.py

```diff
@@ -3247,7 +3247,20 @@ def runtime_wrapper(*args):
                     if trace_joint:
                         assert isinstance(updated_inpt, TensorAlias)
                         updated_inpt = updated_inpt.alias
-                    original_inpt.set_(updated_inpt)
+                    try:
+                        original_inpt.set_(updated_inpt)
+                    except RuntimeError:
+                        # This happens when the sizes of the two tensors do not match, probably
+                        # because we saw .data
+                        # while we could call
+                        # original_inpt.data = updated_inpt
+                        # That's a little gross, let's use the "safe" pattern instead
```

Comment:
I don't really see why there is any benefit to using the "safe" pattern here. You are trying to avoid `.data` setting because it... what... traces improperly? But this is runtime wrapper code, it is never getting traced.

Reply:
Consistency is my only real, weak argument here. We can switch it to .data if you prefer.